java/src/main/kotlin/io/spine/protodata/java/Expression.kt

```diff
@@ -28,364 +28,51 @@
-
-    private fun fieldAccess(fieldName: String) = FieldAccess(this, fieldName)
-
-    /**
-     * Constructs an expression chaining a call of the `build()` method.
-     */
-    public fun chainBuild(): MethodCall =
-        chain("build")
-
-    internal companion object {
-        internal const val OF = "of"
-    }
-}
-
-/**
- * Constructs an expression of a list from the given list of [expressions].
+ * ```
+ * val eightInt = ArbitraryExpression<Int>("4 + 4")
+ * val eightDouble = ArbitraryExpression<Double>("4 + 4")
+ * ```
  *
- * The resulting expression always yields an instance of Guava `ImmutableList`.
- */
-public fun listExpression(expressions: List<Expression>): MethodCall =
-    ClassName(ImmutableList::class).call(OF, expressions)
-
-/**
- * Constructs an expression of a list of the given [expressions].
+ * Pay attention that the expressions with the same code are not differentiated
+ * at runtime.
  *
- * The resulting expression always yields an instance of Guava `ImmutableList`.
- */
-public fun listExpression(vararg expressions: Expression): MethodCall =
-    listExpression(expressions.toList())
-
-/**
- * Constructs an expression of a map of the given [expressions].
+ * If we compare the variables declared in the example above, we will get `true`:
  *
- * The resulting expression always yields an instance of Guava `ImmutableMap`.
+ * ```
+ * println(eightInt.equals(eightDouble)) // Prints `true`.
+ * ```
  *
- * @param expressions The expressions representing the entries.
- * @param keyType The type of the keys in the map;
- *   must be non-`null` if the map is not empty, may be `null` otherwise.
- * @param valueType The type of the values in the map;
- *   must be non-`null` if the map is not empty, may be `null` otherwise.
- */
-public fun mapExpression(
-    expressions: Map<Expression, Expression>,
-    keyType: ClassName?,
-    valueType: ClassName?
-): MethodCall {
-    val immutableMapClass = ClassName(ImmutableMap::class)
-
-    if (expressions.isEmpty()) {
-        return immutableMapClass.call(OF)
-    }
-    checkNotNull(keyType) { "Map key type is not set." }
-    checkNotNull(valueType) { "Map value type is not set." }
-    var call = immutableMapClass.call("builder", generics = listOf(keyType, valueType))
-    expressions.forEach { (k, v) ->
-        call = call.chain("put", listOf(k, v))
-    }
-    return call.chainBuild()
-}
-
-/**
- * Formats these class names as type arguments, including the angle brackets.
- */
-private fun List<ClassName>.genericTypes() =
-    if (isEmpty()) "" else "<${joinToString()}>"
-
-/**
- * Formats these expressions as method parameters, not including the brackets.
+ * @param T The type of the returned value.
+ * @param code The Java code denoting an expression.
  */
-private fun List<Expression>.formatParams() =
-    joinToString { it.toCode() }
+public open class ArbitraryExpression<T>(code: String): ArbitraryElement(code), Expression<T>
```

Comment:
Can we have `AnExpression` and `AnElement` just for brevity?

Reply:
Another question is, why do we need an open class which does not have anything apart from the value?